modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -1237,6 +1237,10 @@ void GDScriptAnalyzer::resolve_class_body(GDScriptParser::ClassNode *p_class, co
 					member.variable->getter->set_datatype(member.variable->datatype);
 
 					resolve_function_body(member.variable->getter);
+
+					if (!member.variable->getter->get_datatype().is_hard_type() && !member.variable->getter->body->has_return) {
+						push_error(R"(Not all code paths return a value.)", member.variable->getter);
+					}
```

Comment:
```suggestion
					if (!member.variable->getter->body->has_return) {
						push_error(R"(Not all code paths in the getter function return a value.)", member.variable->getter);
					}
```

I think we don't care if the variable/getter is typed or not. A getter must always return something. Accordingly, it is worth clarifying the reason in the error message.

Reply:
I tried to remove the first condition but a duplicate error would be generated by the `resolve_function_body` function when analyzing the getter of a typed variable. Maybe we should modify `resolve_function_body` and pass a bool as a parameter indicating whether the function to analyze is a getter?

Reply:
This is not the only case where we have unnecessary errors, see also #75052. If the error order is correct, then I think it's OK for now.